exporter/syslogexporter/config.go

```diff
@@ -21,6 +21,11 @@ var (
 	errOctetCounting       = errors.New("octet counting is only supported for rfc5424 protocol")
 )
 
+const (
+	networkTCP = "tcp"
```

Comment:
confignet exposes these as `TransportTypeTCP` and `TransportTypeUDP`, would it be useful to reuse those?

Reply:
@codeboten @andrzej-stencel I changed the code to use `TransportTypeTCP`  and `TransportTypeUDP` from `confignet`, please take a look.

I considered also changing type of [network](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/2d6c34d57137fc9a5002c15f6354cec434c5c68d/exporter/syslogexporter/config.go#L32) in syslog exporter from `string` to `type TransportType string` from `confignet` but it'll cause unnecessary breaking change - [UnmarshalText](https://github.com/open-telemetry/opentelemetry-collector/blob/dc48d0e66b39dc5a0cd2bac68d964eef70863219/config/confignet/confignet.go#L35) does not accept strings written with using capital letters but current version syslog exporter accepts `network` written in different forms.

Reply:
Awesome, thank you Kasia, and thanks Alex for pointing this out. Let's give Alex a chance to review, it's ready to merge from my perspective.